### ISSUE TITLE
Changing a peername is precarious

### ISSUE THREAD

**Issue by ramfox** [qri-io/qri#335]
When peername is changed, there are a few major areas that are effected:

* The repo: each datasetRef is saved to the store with a peername. If the peername associated with a profileID is changed, we need to go through and reflect this change in each datasetRef.
* A peer's peers list -> if the peername changes, how will peers see the change? Will they look at the profileID, see its the same and update the peername? or will they just add this 'new' peer to the end of their list?
* A dataset's log: a dataset has a history, and each commit in the history is associated with a peer. Will we have to go through each commit in history and change the peername
* A peer's dataset's log: if a peer adds one of my datasets (which should include the history of my dataset), and then I change my peername, that should be reflected in my peer's dataset's commits

**Comment by b5** [comment]
and the award for best issue name goes to...

**Comment by b5** [comment]
Yeah, for now we should disallow changing your peername and after this release immediately set to a series of methods that update a peername change, keeping in mind that **peers will cache this information as well, and will need to be notified of any peername change so they can update their records**.

But yes, for now let's just make it impossible to change your peername

**Comment by ramfox** [comment]
got it!
